cli/azd/cmd/templates.go

```diff
@@ -7,17 +7,27 @@ import (
 	"context"
 	"fmt"
 	"io"
+	"net/http"
 
 	"github.com/azure/azure-dev/cli/azd/cmd/actions"
+	"github.com/azure/azure-dev/cli/azd/pkg/config"
+	"github.com/azure/azure-dev/cli/azd/pkg/input"
 	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/output/ux"
 	"github.com/azure/azure-dev/cli/azd/pkg/templates"
 	"github.com/spf13/cobra"
 )
 
 func templateNameCompletion(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
-	templateManager := templates.NewTemplateManager()
-	templates, err := templateManager.ListTemplates()
+	templateManager, err := templates.NewTemplateManager(
+		templates.NewSourceManager(config.NewUserConfigManager(), http.DefaultClient),
```

Comment:
Shouldn't this method be using the objects from the `container` ?  
Consider making the TemplateManager an input argument for this method